### ISSUE TITLE
Add algorithm options for Kmeans

### ISSUE THREAD

**Issue by tk382** [tk382/HIPPO#2]
To avoid the following Quick TRANSFERing error https://stackoverflow.com/questions/21382681/kmeans-quick-transfer-stage-steps-exceeded-maximum
add user-option to choose "MacQueen" or "Lloyd".

**Comment by tk382** [comment]
Also, add functionalities for different clustering methods than K-means. For example, Louvian or Leiden.

**Comment by shanzhyang** [comment]
Hi, i found only "Seurat" and "SC3" can be found as options of the clustering methods. So the Seurat is louvian and SC3 is Leiden? Thanks!

**Comment by tk382** [comment]
Yes - the original hippo function uses 'PCA+kmeans' (parameter 'kmeans'), but it can also directly calls the function from the Seurat and SC3 package that use Louvain and Leiden respectively. Thanks.
